plugins/languages/treesitter/treesitter.nix

```
@@ -6,210 +6,306 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.treesitter;
-in
-{
-  options = {
-    plugins.treesitter = {
-      enable = mkEnableOption "tree-sitter syntax highlighting";
-
-      package = mkOption {
-        type = types.package;
-        default = pkgs.vimPlugins.nvim-treesitter;
-        description = "Plugin to use for nvim-treesitter. If using nixGrammars, it should include a `withPlugins` function";
-      };
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "treesitter";
+  originalName = "nvim-treesitter";
+  luaName = "nvim-treesitter.configs";
+  defaultPackage = pkgs.vimPlugins.nvim-treesitter;
 
-      nixGrammars = mkOption {
-        type = types.bool;
-        default = true;
-        description = "Install grammars with Nix";
-      };
+  maintainers = [ ];
```

We need a maintainer.

---

```suggestion
  maintainers = [ lib.maintainers.khaneliman ];
```

Not sure if I missed this or it got mistakenly reverted?